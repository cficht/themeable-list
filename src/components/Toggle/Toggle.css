.Toggle {
  display: flex;
  align-items: center;

  span {
    margin: 0px 15px 0px 15px;
  }
  
  input {
    display: none;
  }

  label {
    display: block;
    width: 50px;
    height: 25px;
    border: 1px solid black;
    background-color: white;
  }

  label::before {
    content: '';
    display: block;
    width: 25px;
    height: 25px;
    background-color: black;
    transition: 0.2s;
  }

  input:checked + label {
    background-color: white;
    &::before {
      margin-left: 25px;
    }
  }
}
