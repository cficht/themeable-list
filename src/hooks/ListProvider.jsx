import React, { createContext, useState, useEffect, useContext } from 'react';
import PropTypes from 'prop-types';

const ListContext = createContext();

export const ListProvider = ({ children }) => {
  const [characters, setCharacters] = useState([]);

  const chararcterTest = [
    {
      id: 1,
      name: 'Rick Sanchez',
      status: 'Alive',
      species: 'Human',
      image: 'https://rickandmortyapi.com/api/character/avatar/1.jpeg'
    },
    {
      id: 2,
      name: 'Morty Smith',
      status: 'Alive',
      species: 'Human',
      image: 'https://rickandmortyapi.com/api/character/avatar/2.jpeg'
    },
    {
      id: 3,
      name: 'Summer Smith',
      status: 'Alive',
      species: 'Human',
      image: 'https://rickandmortyapi.com/api/character/avatar/3.jpeg'
    }
  ];

  useEffect(() => {
    setCharacters(chararcterTest);
  }, []);

  return (
    <ListContext.Provider value={{ characters }}>
      {children}
    </ListContext.Provider>
  );
};

ListProvider.propTypes = {
  children: PropTypes.node
};

export const useCharacters = () => {
  const { characters } = useContext(ListContext);
  return characters;
};
