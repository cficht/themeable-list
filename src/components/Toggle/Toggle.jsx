import React from 'react';
import styles from './toggle.css';
import { useStatus, useToggle } from '../../hooks/ListProvider';

const Toggle = () => {
  const status = useStatus();
  const toggle = useToggle();

  return (
    <section className={styles.Toggle}>
      <span>DEAD</span>
      <input id="toggle" type="checkbox" checked={status === 'alive'} onChange={toggle} />
      <label htmlFor="toggle"></label>
      <span>ALIVE</span>
    </section>
  );
};

export default Toggle;
