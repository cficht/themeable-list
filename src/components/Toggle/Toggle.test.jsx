import React from 'react';
import { shallow } from 'enzyme';
import Toggle from './Toggle';
import { ListProvider } from '../../hooks/ListProvider';

describe('Toggle component', () => {
  it('renders Toggle', () => {
    const wrapper = shallow(
      <ListProvider>
        <Toggle />
      </ListProvider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
