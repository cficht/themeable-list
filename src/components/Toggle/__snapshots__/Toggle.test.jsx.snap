// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle component renders Toggle 1`] = `
<ContextProvider
  value={
    Object {
      "characters": Array [],
      "handlePage": [Function],
      "page": 1,
      "status": "alive",
      "toggle": [Function],
    }
  }
>
  <Toggle />
</ContextProvider>
`;
